apps/survey/contracts/Surveying.sol

```diff
@@ -0,0 +1,256 @@
+        minAcceptParticipationPct = _minAcceptParticipationPct;
+        surveyTime = _surveyTime;
+    }
+
+    /**
+    * @notice Change minimum acceptance participation to `(_minAcceptParticipationPct - _minAcceptParticipationPct % 10^14) / 10^16`%
+    * @param _minAcceptParticipationPct New acceptance participation
+    */
+    function changeMinAcceptParticipationPct(uint256 _minAcceptParticipationPct) authP(MODIFY_PARTICIPATION_ROLE, arr(_minAcceptParticipationPct)) external {
+        require(_minAcceptParticipationPct > 0);
+        require(_minAcceptParticipationPct <= PCT_BASE);
+        minAcceptParticipationPct = _minAcceptParticipationPct;
+
+        ChangeMinParticipation(_minAcceptParticipationPct);
+    }
+
+    /**
+    * @notice Create a new non-binding survey about "`_metadata`"
+    * @param _metadata Survey metadata
+    * @param _options Number of options voters can decide between
+    * @return surveyId id for newly created survey
+    */
+    function newSurvey(string _metadata, uint256 _options) auth(CREATE_SURVEYS_ROLE) external returns (uint256 surveyId) {
+        return _newSurvey(_metadata, _options);
+    }
+
+    /**
+    * @notice Vote for multiple options in survey #`_surveyId`.
+    * @dev Empty options and stakes arrays allow to remove previous votes
+    * @param _surveyId Id for survey
+    * @param _options Array with indexes of supported options
+    * @param _stakes Number of tokens assigned to each option
+    */
+    function voteOptions(uint256 _surveyId, uint256[] _options, uint256[] _stakes) public {
+        require(_options.length == _stakes.length);
+        require(canVote(_surveyId, msg.sender));
+
+        uint256 i;
+        Survey storage survey = surveys[_surveyId];
+
+        uint256 voterStake = token.balanceOfAt(msg.sender, survey.snapshotBlock);
+        MultiOptionVote storage previouslyVoted = survey.voters[msg.sender];
+
+        // revert previous votes, if any
+        if (previouslyVoted.optionsCastedLength > 0) {
+            // voter removes their vote
+            for (i = 1; i <= previouslyVoted.optionsCastedLength; i++) {
+                survey.votes[previouslyVoted.castedVotes[i].optionId] = survey.votes[previouslyVoted.castedVotes[i].optionId].sub(previouslyVoted.castedVotes[i].stake);
+            }
+            // remove previous vote participation
+            survey.participation = survey.participation.sub(voterStake.sub(previouslyVoted.castedVotes[0].stake));
+
+            // reset previously voted options, if any
+            delete survey.voters[msg.sender];
+        }
+
+        uint256 totalVoted = 0;
+        // reserve first index for NO_VOTE
+        survey.voters[msg.sender].castedVotes[0] = OptionCast({optionId: NO_VOTE, stake: 0});
+        for (i = 0; i < _options.length; i++) {
+            // index 0 in options array is reserved for NO_VOTE, so index will be 1 more
+            _voteOption(_surveyId, i+1, _options[i], _stakes[i]);
```

Comment:
`i + 1`

Reply:
You can just generate the `OptionCast` instance here and past that to the internal function